src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/Scripts.cs

```diff
@@ -41,5 +43,29 @@ public Scripts(Compiler compiler)
             }
             return null;
         }
+
+        internal class TrimSafeDictionary
+        {
+            private readonly Dictionary<string, Type?> _backingDictionary = new Dictionary<string, Type?>();
+
+            public Type? this[string key]
+            {
+                [UnconditionalSuppressMessage("TrimAnalysis", "IL2073:MissingDynamicallyAccessedMembers",
+                    Justification = "The getter of the dictionary is not annotated to preserve the constructor, but the sources that are adding the items to " +
+                    "the dictionary are annotated so we can supress the message as we know the constructor will be preserved.")]
+                [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
```

Comment:
Can we just put the attribute on the whole property?

Reply:
Chatted offline, this is a problem specific to Indexers and have logged the following issue for investigation tracking: https://github.com/mono/linker/issues/1902